### ISSUE TITLE
Invalid URI strings generated in the ErrorLog output

### ISSUE THREAD

**Issue by mavasani** [dotnet/roslyn#10349]
**Version Used**: VS2015 Update2 (1.2.0 compiler toolset)

**Steps to Reproduce**:

1. git clone https://github.com/jamesNK/Newtonsoft.Json/
2. Edit `<%root%>\newtonsoft.json\Newtonsoft.Json\Newtonsoft.Json.Portable.csproj` to add the ErrorLog property to the top level property group: `<ErrorLog>ErrorLog.json</ErrorLog>`
3. Open this project in VS
4. Build the project

**Expected Behavior**:
Valid URI strings in the error log file.

**Actual Behavior**:
URI strings in the emitted ErrorLog.json file seem to have `\/` character sequence, seems like escaping is broken. For example:
```
"uri": "file:\/\/\/C:\/Newtonsoft.Json\/Newtonsoft.Json\/Bson\/BsonReader.cs",
```
I switched to using the 1.1.1 compiler toolset for building and file path with correct escape character seems to be emitted `C:\\Newtonsoft.Json\\Newtonsoft.Json\\Bson\\BsonReader.cs`.

**Comment by mavasani** [comment]
@nguerrera may have context on this.

**Comment by nguerrera** [comment]
@mavasani This is not invalid, just not pretty. ;) JSON permits escaping forward slashes this way (it is *optional* per spec) and the DataContract serializer used to do this on our behalf and out of our control. I have since lifted the dependency on DataContractSerializer but retained the escaping logic. I made a note to tidy this up and it's already tracked by #9769. I think this can be closed as dup of that.
